numpy/lib/shape_base.py

```diff
@@ -89,9 +90,20 @@ def apply_along_axis(func1d, axis, arr, *args, **kwargs):
     outshape = asarray(arr.shape).take(indlist)
     i.put(indlist, ind)
     res = func1d(arr[tuple(i.tolist())], *args, **kwargs)
-    #  if res is a number, then we have a smaller output array
+
+    # see gh-8363
+    _strcase = False
+    _dtypes = set()
+
+    # If res is a number, then we have a smaller output array.
     if isscalar(res):
```

Comment:
@shoyer : Okay, understanding more...what would the return `dtype` of `apply_along_axis` be in this case?  Also, if you don't mind explaining, what is your rationale for forcing the user to do this?